app/data_models/progress_store.py

```diff
@@ -170,45 +152,39 @@ def update_section_or_repeating_blocks_progress_completion_status(
         return updated
 
     def get_section_or_repeating_blocks_progress_status(
-        self, section_id: str, list_item_id: str | None = None
+        self, section_key: SectionKey
     ) -> str:
         """
         Return the CompletionStatus of the Section or Repeating Blocks for a list item,
         specified by the given section_id and list_item_id.
         Returns NOT_STARTED if the progress does not exist
         """
-        progress_key = SectionKey(section_id, list_item_id)
-        if progress_key in self._section_and_repeating_blocks_progress:
-            return self._section_and_repeating_blocks_progress[progress_key].status
+        progress_key = section_key
```

Comment:
No need to assign this var, can just use section_key

Reply:
I was not sure about touching it but it makes sense to remove this, done, thanks!